cnd/src/main.rs

```diff
@@ -142,6 +142,13 @@ fn main() -> anyhow::Result<()> {
     };
 
     runtime.block_on(load_swaps::load_swaps_from_database(deps.clone()))?;
+
+    let listen_addr = settings.http_api.socket;
+    // This is racey but gives us some protection, if the race condition is hit it
+    // just means cnd will start even though the HTTP server did not come up.
+    if !port_is_available(listen_addr.port()) {
+        return Err(anyhow::anyhow!("HTTP port is unavailable: {}", listen_addr));
+    }
     runtime.spawn(spawn_warp_instance(settings, deps));
```

Comment:
```suggestion
    runtime.block_on(spawn_warp_instance(settings, deps));
```
This one liner change is all you need. (and delete `::std::thread::park();`

Reply:
Cool, cheers.  Implemented as suggested.